libs/src/api/account.js

```diff
@@ -328,6 +329,58 @@ class Account extends Base {
     this.REQUIRES(Services.DISCOVERY_PROVIDER)
     return this.discoveryProvider.searchTags(text, user_tag_count, kind, limit, offset)
   }
+
+  /**
+   * Check if the user has a distribution claim
+   * @param {number?} index The index of the claim to check (if known)
+   */
+  async getHasClaimed (index) {
+    this.REQUIRES(Services.DISCOVERY_PROVIDER)
+    if (index) {
+      return this.ethContracts.ClaimDistributionClient.isClaimed(index)
+    }
+    const userWallet = this.web3Manager.getWalletAddress()
+    const claim = await this.comStock.getComStock({ wallet: userWallet })
+    return this.ethContracts.ClaimDistributionClient.isClaimed(claim.index)
+  }
+
+  /**
+   * Get the distribution claim amount
+   */
+  async getClaimDistributionAmount () {
+    this.REQUIRES(Services.COM_STOCK)
+    const userWallet = this.web3Manager.getWalletAddress()
+    const claimDistribution = await this.comStock.getComStock({ wallet: userWallet })
+    const amount = Utils.toBN(claimDistribution.amount) 
+    return amount
+  }
+
+  /**
+   * Make the claim
+   * @param {number?} index The index of the claim to check
+   * @param {BN?} amount The amount to be claimed
+   * @param {Array<string>?} merkleProof The merkle proof for the claim
+   */
+  async makeDistributionClaim (index, amount, merkleProof) {
+    this.REQUIRES(Services.COM_STOCK)
```

Comment:
should this also require identity? cause of relay

Reply:
yeah, good call